diff -Nur Python-2.7.5/setup.py Python-2.7.5.minitage/setup.py
--- Python-2.7.5/setup.py	2013-05-12 05:32:54.000000000 +0200
+++ Python-2.7.5.minitage/setup.py	2013-05-30 17:52:23.389920810 +0200
@@ -496,7 +496,10 @@
         # if a file is found in one of those directories, it can
         # be assumed that no additional -I,-L directives are needed.
         inc_dirs = self.compiler.include_dirs[:]
-        lib_dirs = self.compiler.library_dirs[:]
+        mlib_dirs = []
+        for i in os.environ.get('LD_RUN_PATH', '').split(':'):
+            mlib_dirs.append(i)
+        lib_dirs = mlib_dirs + self.compiler.library_dirs[:]
         if not cross_compiling:
             for d in (
                 '/usr/include',
@@ -737,8 +740,9 @@
                 # MacOSX 10.4 has a broken readline. Don't try to build
                 # the readline module unless the user has installed a fixed
                 # readline package
-                if find_file('readline/rlconf.h', inc_dirs, []) is None:
-                    do_readline = False
+                #if find_file('readline/rlconf.h', inc_dirs, []) is None:
+                #    do_readline = False
+                print 'readline module doesn\'t compile out of the box except inside minitage :).' 
         if do_readline:
             if host_platform == 'darwin' and os_release < 9:
                 # In every directory on the search path search for a dynamic
@@ -768,7 +772,8 @@
 
         # crypt module.
 
-        if self.compiler.find_library_file(lib_dirs, 'crypt'):
+#        if self.compiler.find_library_file(lib_dirs, 'crypt'):
+        if host_platform not in ['compile otb on minitage :)']:
             libs = ['crypt']
         else:
             libs = []
@@ -802,8 +807,9 @@
         if (ssl_incs is not None and
             ssl_libs is not None):
             exts.append( Extension('_ssl', ['_ssl.c'],
-                                   include_dirs = ssl_incs,
-                                   library_dirs = ssl_libs,
+                                   # they are already in compilation flags when building with minitage
+                                   #include_dirs = ssl_incs,
+                                   #library_dirs = ssl_libs, 
                                    libraries = ['ssl', 'crypto'],
                                    depends = ['socketmodule.h']), )
         else:
@@ -1074,9 +1080,10 @@
             # is on an NFS server that goes away).
             exts.append(Extension('_bsddb', ['_bsddb.c'],
                                   depends = ['bsddb.h'],
-                                  library_dirs=dblib_dir,
-                                  runtime_library_dirs=dblib_dir,
-                                  include_dirs=db_incs,
+                                  # defined by minitage
+                                  #library_dirs=dblib_dir,
+                                  #runtime_library_dirs=dblib_dir,
+                                  #include_dirs=db_incs, 
                                   libraries=dblibs))
         else:
             if db_setup_debug: print "db: no appropriate library found"
@@ -1362,8 +1369,10 @@
             missing.append('_curses')
 
         # If the curses module is enabled, check for the panel module
-        if (module_enabled(exts, '_curses') and
-            self.compiler.find_library_file(lib_dirs, panel_library)):
+        if (module_enabled(exts, '_curses')): # always true on minitage and
+            #self.compiler.find_library_file(lib_dirs, panel_library)): 
+#        if (module_enabled(exts, '_curses') and
+#            self.compiler.find_library_file(lib_dirs, panel_library)):
             exts.append( Extension('_curses_panel', ['_curses_panel.c'],
                                    libraries = [panel_library] + curses_libs) )
         else:
@@ -1480,8 +1489,12 @@
 
         exts.append(Extension('pyexpat',
                               define_macros = define_macros,
-                              include_dirs = expat_inc,
-                              libraries = expat_lib,
+                              # defined by minitage
+                              # include_dirs = [expatinc],
+                              # force link against minitage expat
+                              libraries = ['expat'], 
+#                              include_dirs = expat_inc,
+#                              libraries = expat_lib,
                               sources = ['pyexpat.c'] + expat_sources,
                               depends = expat_depends,
                               ))
@@ -1493,8 +1506,12 @@
             define_macros.append(('USE_PYEXPAT_CAPI', None))
             exts.append(Extension('_elementtree',
                                   define_macros = define_macros,
-                                  include_dirs = expat_inc,
-                                  libraries = expat_lib,
+                                  # defined by minitage
+                                  # include_dirs = [expatinc],
+                                  ## force link against minitage expat
+                                  libraries = ['expat'],  
+                                  #include_dirs = expat_inc,
+                                  #libraries = expat_lib,
                                   sources = ['_elementtree.c'],
                                   depends = ['pyexpat.c'] + expat_sources +
                                       expat_depends,
