diff -Nur Python-2.7.5/setup.py Python-2.7.5.tk/setup.py
--- Python-2.7.5/setup.py	2013-05-12 05:32:54.000000000 +0200
+++ Python-2.7.5.tk/setup.py	2013-05-30 17:34:32.368981884 +0200
@@ -1722,6 +1722,7 @@
         return missing
 
     def detect_tkinter_darwin(self, inc_dirs, lib_dirs):
+        return 1
         # The _tkinter module, using frameworks. Since frameworks are quite
         # different the UNIX search logic is not sharable.
         from os.path import join, exists
@@ -1805,6 +1806,7 @@
 
 
     def detect_tkinter(self, inc_dirs, lib_dirs):
+        return 1
         # The _tkinter module.
 
         # Rather than complicate the code below, detecting and building
